src/main/java/org/apache/accumulo/testing/continuous/AsyncRecordReader.java

```diff
@@ -0,0 +1,85 @@
+package org.apache.accumulo.testing.continuous;
+
+import java.util.Map;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.LongAdder;
+import java.util.stream.IntStream;
+import java.util.zip.CRC32;
+
+import org.apache.accumulo.core.data.Value;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+import com.google.common.collect.Maps;
+
+public class AsyncRecordReader extends ContinousRecordReader {
+
+  int threads = 0;
+  int queuedKeys = 100;
+
+  ExecutorService service = null;
+
+  BlockingQueue<Map.Entry<BulkKey,Value>> queue;
+
+  final LongAdder generatedCount = new LongAdder();
+  final AtomicBoolean running = new AtomicBoolean(false);
+
+  @Override
+  public void initialize(InputSplit inputSplit, TaskAttemptContext job) {
+    super.initialize(inputSplit, job);
+
+    threads = job.getConfiguration().getInt(ContinousInputOptions.PROP_ASYNC, 5);
+    queuedKeys = job.getConfiguration().getInt(ContinousInputOptions.PROP_ASYNC_KEYS, 1000);
+    queue = new ArrayBlockingQueue<>(queuedKeys);
+
+    service = Executors.newFixedThreadPool(threads);
+  }
+
+  @Override
+  public boolean nextKeyValue() {
+    currKey = null;
+    if (nodeCount < numNodes) {
+      if (!running.get()) {
+        running.set(true);
+        // start up lazily
+        IntStream.range(0, threads).forEach(x -> {
+          Runnable generator = () -> {
+            CRC32 cksum = checksum ? new CRC32() : null;
+            while (generatedCount.longValue() < numNodes && running.get()) {
+              try {
+                final BulkKey key = genKey(cksum);
+                final Value value = new Value(createValue(uuid, key.getRowData().toArray(), cksum));
+                while (!queue.offer(Maps.immutableEntry(key, value), 1, TimeUnit.SECONDS)) {
+                  if (!running.get())
```

Comment:
Does anything ever set this to false after its been set to true?

Reply:
Let me double check. Thanks!

Reply:
I see a test failure so I'll check those too. Thanks for the review!